autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/SectionSelect.tsx

```diff
@@ -93,31 +108,123 @@ const SectionSelect: React.FC<SectionSelectProps> = ({ id }): JSX.Element => {
+        onClose={(): void => {
+          setSortState({ ...sortState, sortMenuAnchor: null });
+        }}
+      >
+        <MenuItem
+          onClick={(): void => { handleChange(SortType.DEFAULT); }}
+          selected={reduxSortType === SortType.DEFAULT}
+        >
+          Default
+        </MenuItem>
+        <MenuItem
+          onClick={(): void => { handleChange(SortType.SECTION_NUM); }}
+          selected={reduxSortType === SortType.SECTION_NUM}
+        >
+          Section Num
+        </MenuItem>
+        <MenuItem
+          onClick={(): void => { handleChange(SortType.GRADE); }}
+          selected={reduxSortType === SortType.GRADE}
+        >
+          Grade
+        </MenuItem>
+        <MenuItem
+          onClick={(): void => { handleChange(SortType.INSTRUCTOR); }}
+          selected={reduxSortType === SortType.INSTRUCTOR}
+        >
+          Instructor
+        </MenuItem>
+        <MenuItem
+          onClick={(): void => { handleChange(SortType.OPEN_SEATS); }}
+          selected={reduxSortType === SortType.OPEN_SEATS}
+        >
+          Open Seats
+        </MenuItem>
+        <MenuItem
+          onClick={(): void => { handleChange(SortType.HONORS); }}
+          selected={reduxSortType === SortType.HONORS}
+        >
+          Honors
+        </MenuItem>
+      </Menu>
+    </>
+  );
+
+  // Select All
   // pre-making list so we can tell if the select-all checkbox should be checked
   const list = makeList();
   const allSelected: boolean = numSelected === sections.length;
+  const selectAll = (
+    <ToggleButton classes={{ root: classes.rootToggleButton }} value="select-all" aria-label="select all" onChange={(): void => { dispatch(toggleSelectedAll(id, !allSelected)); }}>
+      <Checkbox
+        checked={allSelected}
+        value={(allSelected) ? 'allOn' : 'allOff'}
+        color="primary"
+        size="small"
+        disableRipple
+        classes={{ root: classes.rootCheckbox }}
+      />
+        SELECT ALL
+    </ToggleButton>
+  );
+
+  // div of options for easier version control
   const sectionSelectOptions = (
     <div>
-      <ToggleButton classes={{ root: classes.rootToggleButton }} value="select-all" aria-label="select all" onChange={(): void => { dispatch(toggleSelectedAll(id, !allSelected)); }}>
-        <Checkbox
-          checked={allSelected}
-          value={(allSelected) ? 'allOn' : 'allOff'}
-          color="primary"
-          size="small"
-          disableRipple
-          classes={{ root: classes.rootCheckbox }}
-        />
-          SELECT ALL
-      </ToggleButton>
+      {selectAll}
+      {sortMenu}
     </div>
   );
 
+  // don't show loading for small number of sections since its almost instant
+  // and causes ugly flashing
   return (
     <>
       {sectionSelectOptions}
-      <List disablePadding className={styles.sectionRows}>
-        {list}
-      </List>
+      {(sortState.frontendSortType === reduxSortType || sections.length <= 4) ? (
```

Comment:
Seems like most classes have a number of sections either much lower or much higher than 4. 
In addition, if we show the loading for sections lower than 4, it expands the card since there's no way of knowing how much space the sections will take. This happens for like one frame which kinda hurts the eyes. 4 is the threshold I've found for it taking just long enough that the loading indicator doesn't flash, there's enough time to read it and also 4 sections typically results in a max size card.